src/loader/bgw_counter.c

```diff
@@ -124,8 +124,13 @@ bgw_total_workers_decrement()
 	if (ct->total_workers > 1)
 		ct->total_workers--;
 	else
+	{
+		SpinLockRelease(&ct->mutex);
 		ereport(ERROR, (errmsg("TimescaleDB background worker internal error: cannot decrement workers below 1"),
 						errhint("The background worker scheduler is in an invalid state, please submit a bug report.")));
+	}
+
+
 	SpinLockRelease(&ct->mutex);
```

Comment:
Just for readability, can you move this `SpinLockRelease` inside the `if`? Then it becomes clear that this is not a viable code path if you went down the `else` branch. Similarly below.

Reply:
I think I agree with @amytai here. Probably will prevent someone from removing it from the `else` in the future.

Reply:
Done